server/asset/eth/eth_test.go

```diff
@@ -498,3 +499,28 @@ func TestToCoinID(t *testing.T) {
 		}
 	}
 }
+
+// TestRequiredOrderFunds ensures that a fee calculation in the calc package
+// will come up with the correct required funds.
+func TestRequiredOrderFunds(t *testing.T) {
+	eth := new(Backend)
+	swapVal := uint64(1000000000)                // gwei
+	numSwaps := uint64(17)                       // swaps
+	initSizeBase := uint64(eth.InitTxSizeBase()) // 0 gas
+	initSize := uint64(eth.InitTxSize())         // init value gas
+	feeRate := uint64(30)                        // gwei
+
+	// We want the fee calculation to simply be the gas used for each swap
+	// plus the initial value.
```

Comment:
Clear enough, but would it be more accurate to say "**the cost** of the gas used for each swap", because it's gas*gas_price = gas * (gwei / gas) = gwei.

In that vein, I feel like it could provide more clarity with `feeRate` line comment as `// gwei / gas`, even though just `gwei` seems to be most common.

Reply:
Wait, what was the question. lol. Changed the wording in the test. Was there another place?

It looks like I did make that distinction in the `backend.FeeRate` method doc. 

Gas is a fixed unit in the evm and certain actions cost a certain amount of it. It doesn't care about supply/demand. the gasPrice is how much one gas costs, and is wei or gwei per gas. It is variable depending on market conditions. Most recently 100 gwei per gas.

I'm sure you understood already, just reinterating your reinteration. gasPrice is the cost of a gas unit.

Reply:
> the gasPrice is how much one gas costs, and is wei or gwei per gas

Just this.  You say "fee calculation to simply be the gas used for each swap plus" but it's gas * feeRate plus...

Yeah, the change in https://github.com/decred/dcrdex/pull/1157/commits/c60a4e29db37f70e474f810e459b4c9a29cc2c07 is exactly what I was getting at, that gas cost is (gwei / gas) regardless of what websites say, and that why the units work out when multiplying by gas.